docs/features/deconstruction.md

```diff
@@ -100,7 +102,17 @@ evaluate the right-hand-side and do a tuple conversion (using a fake tuple repre
 assign element-wise from the right to the left
 ```
 
-Note that tuples (`System.ValueTuple` and `System.Tuple`) don't need to invoke Deconstruct.
+#### Resolution of the Deconstruct method
+
+The resolution is equivalent to typing `rhs.Deconstruct(out var x1, out var x2, ...);` with the appropriate number of parameters to deconstruct into.
+It is based on normal overload resolution.
+This implies that `rhs` cannot be dynamic.
```

Comment:
I am not sure if the Out Var limitation should necessarily translate into limitation for Deconstruction. I think it would be reasonable to treat all the output values as dynamic.